### DISCUSSION TITLE
Which .NET versions should AutoFixture support?

### DISCUSSION THREAD

**Discussion by aivascu** [AutoFixture/AutoFixture#1442] [Polls]
With the current release schedule of .NET and by extension C# (thanks Microsoft) the number of things that AutoFixture should support keep piling up. Adding them all will effectively create different versions of the library depending on what version is targeted in the test project.

Cumulatively adding .NET versions to the library, implies adding conditional compilation directives (e.g. `#if NET5_0_OR_GREATER`), both in the library and in the test projects, as well as in the targeted packages.
This will create the situation where the library might change behavior/requirements depending on the target framework, especially inconvenient for code-bases that use multi-targeting. This will also slow down development of the library itself, since package dependencies will have to be managed for each target framework.

The amount of complexity can be limited if AutoFixture will support fewer version of .NET instead of targeting each version that introduces a new feature. This however will mean that the library will lock out from patches anyone that is still targeting an older version of .NET. This will be especially significant for corporate environments that have a lot of code that's adopting new versions slower.

The purpose of this poll is to find out what behavior the uses/community expects going forward.

**Comment by Kralizek** [comment]
I'd add an option: latest major version, regardless if it's current or lts.

Also, I'd consider pairing the version number to that of .NET and aligning the major version with the latest major. That gives a clear sign on what's supported on that release.

If you feel nice you could publish the packages targeting the latest major and the currently supported runtimes but still focusing the latest major.

In this way, people who want to experiment using shims can try using a higher AF version on a earlier runtime.

**Reply by aivascu** [reply]
@Kralizek the option you mentioned is the "Supported LTS and Current (STS)" since according to Microsoft's schedule there cannot be two STS versions supported at the same time. 🙂
The version pairing sounds interesting as well, and it's what I was going for, but now I'm starting to wonder if it wouldn't save time to just go all in with this next release.

**Reply by Kralizek** [reply]
I don't follow your math.

Each LTS has 3 years of support, and we get a new LTS every 2 years, so we have windows where we have 2 LTS supported, like right now with 6 and 8 being supported.

**Reply by aivascu** [reply]
@Kralizek no, I said STS, not LTS. STS versions are supported for 18 months, so the previous STS will be out of support before the next STS arrives.
